Fix arm platform matching

dmcgowan opened the pull request:
The normalization was being inconsistently applied causing a failure to match some platforms in manifest lists. Fix the matcher and normalization to be more consistent and add changes to parser to prevent the defaulted variants from being set in the platform structure.

Fixes #2409

dmcgowan commented:
@stevvooe would this be safe to cherry-pick to 1.1.x?

stevvooe commented:
I didn't think a variant was required for arm64. Perhaps, the bug is in https://github.com/containerd/containerd/blob/master/platforms/cpuinfo.go#L84?

dmcgowan commented:
> I didn't think a variant was required for arm64

It is not required. The normalized form will always be specific now and I updated the `Parse` to not just directly use the normalized form if the value can already be implied and it was given.

I updated to handle the `8` case like in that link and ensured we aren't overwriting a variant we just don't know about yet (someday `arm64/v9`?)

tianon commented:
Looks sane to me 👍
